### DISCUSSION TITLE
Can't salvage volumes located on cordoned node.

### DISCUSSION THREAD

**Discussion by bdkuhman** [longhorn/longhorn#9894] [General]
Noticed this behavior earlier, Don't have the setup for a support bundle or logs anymore, but wanted to post if anyone else has seen this, or if it's expected. 

I was performing maintenance on a few nodes, which unfortunately required rebuilding a few of them. One node in particular was cordoned the whole time and not affected.
After all nodes were up and running, with the single node cordoned; volumes would fail to be salvaged. 
I didn't notice any errors in the k8s events other than the attachdetach controller failed, and didn't notice anything in the manager logs. 

Both autosalvage and manual salvages would fail repeatedly, but after I uncordoned the node, All volumes were salvaged successfully.

I believe all volumes had a replica on this cordoned node, but even if there was an additional failed replica on another node, it would not be salvaged if located on the cordoned node. 

I would think that longhorn should ignore the cordon status for salvages, or at least give out a more detailed/obvious error.

**Comment by ChanYiLin** [comment]
Hi @c3y1huang 
Do you have any idea for this request?

**Comment by voarsh2** [comment]
Cordoned nodes are unable to spin up new pods right? This behaviour is expected?
